autopush/router/simple.py

```diff
@@ -70,6 +78,12 @@ def route_notification(self, notification, uaid_data):
         # Determine if they're connected at the moment
         node_id = uaid_data.get("node_id")
         uaid = uaid_data["uaid"]
+        self.udp = None
+        try:
+            self.udp = json.loads(uaid_data["udp"])
```

Comment:
The common path for 99% of clients is going to be no UDP, exceptions are cheap in Python so we're essentially slow-pathing the common case by adding an exception to it. A 'if udp in uaid_data' test would be fast.